clients/lightrider/lightrider-entry.js

```diff
@@ -54,15 +55,23 @@ async function runLighthouseInLR(connection, url, flags, lrOpts) {
     const runnerResult = await lighthouse(url, flags, config, connection);
     if (!runnerResult) throw new Error('Lighthouse finished without a runnerResult');
 
-    // When LR is called with |internal: {keep_raw_response: true, save_lighthouse_assets: true}|,
-    // this code will log artifacts to raw_response.artifacts.
+    // pre process the LHR for proto
+    const preprocessedLhr = preprocessor.processForProto(runnerResult.lhr);
+
     if (logAssets) {
-      // @ts-ignore - piggyback the artifacts on the LHR.
-      runnerResult.lhr.artifacts = runnerResult.artifacts;
-    }
+      // When LR is called with |internal: {keep_raw_response: true, save_lighthouse_assets: true}|,
+      // this code will log artifacts to raw_response.artifacts.
 
-    // pre process the LHR for proto
-    return JSON.stringify(preprocessor.processForProto(runnerResult.lhr));
+      // Properly serialize artifact errors.
+      const artifactsJson = JSON.stringify(runnerResult.artifacts, assetSaver.stringifyReplacer);
+
+      return JSON.stringify({
+        ...preprocessedLhr,
+        artifacts: JSON.parse(artifactsJson),
+      });
+    } else {
```

Comment:
Can't decide if the else is necessary or not.  I'm flip flopping on keeping it in/removing it is more readable.

Reply:
> Can't decide if the else is necessary or not.  I'm flip flopping on keeping it in/removing it is more readable.

I'd fall on the side of no `else`, but really up to @connorjclark, it's fine either way

Reply:
can you be more specific about what you are suggesting instead

Reply:
You're returning in both sides of the if/else.  You can just remove the else here to always return `JSON.stringify(preprocessedLhr)`.

Reply:
do you see what i'm doing with artifacts tho.

Reply:
oh i see now